server/src/main/java/io/crate/expression/scalar/cast/ImplicitCastFunction.java

```diff
@@ -47,22 +52,41 @@ public static void register(ScalarFunctionModule module) {
                 DataTypes.STRING.getTypeSignature(),
                 DataTypes.UNDEFINED.getTypeSignature()
             ).withTypeVariableConstraints(typeVariable("E")),
-            ImplicitCastFunction::new
+            (signature, boundSignature) -> new ImplicitCastFunction(signature, boundSignature, null)
         );
     }
 
     private final Signature signature;
     private final Signature boundSignature;
+    @Nullable
```

Comment:
I am not a big fan of a nullable final field resulting in a Implicit Function which needs to have a compile call first. However, this is at least immutable and therefore threadsafe.